src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -255,6 +264,26 @@ func parseQuery(r *http.Request) (string, error) {
 	return queries[0], nil
 }
 
+func parseLookbackDuration(r *http.Request, step time.Duration) (time.Duration, bool, error) {
+	lookback := r.FormValue(lookbackParam)
+	if lookback == "" {
+		return 0, false, nil
+	}
+
+	if lookback == stepParam {
```

Comment:
Curious if lookbackDuration is > stepSize should it default to stepSize?

Reply:
I think this is for the special case of `&lookback=step`; although it reads a little weird since it's being compared to a variable named `xxxParam`